File: HexAssetAPI/Models/AssetRequest.cs
using System.ComponentModel.DataAnnotations;

namespace HexAsset.Models
{
	public class AssetRequest
	{
		[Key]
		public int AssetRequestId { get; set; }
		public int AssetId { get; set; }
		public int UserId { get; set; }

		public string? Item {  get; set; } 
		public required string RequestStatus { get; set; }
		public DateTime RequestDate { get; set; } = DateTime.UtcNow;


		public Asset? Asset { get; set; }
		public User? User { get; set; }
	}
}

File: HexAssetUI/src/app/models/AssetRequest.ts
export class AssetRequestModel {
  assetRequestId: number;
  assetId: number;
  userId: number;
  userName: string;
  item: string;
  requestStatus: string;
  requestDate: Date;

  constructor() {
    this.assetRequestId = 0;
    this.assetId = 0;
    this.userId = 0;
    this.userName = "";
    this.item = "";
    this.requestStatus = "";
    this.requestDate = new Date();
  }
}